dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/api/RequestApi.kt

```diff
@@ -0,0 +1,95 @@
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+
+/**
+ * Defines the restful dataland-community-manager API regarding.
+ */
+@RequestMapping("/requests")
+@SecurityRequirement(name = "default-bearer-auth")
+@SecurityRequirement(name = "default-oauth")
+interface RequestApi {
+
+    /**
+     * A method to post a bulk request to Dataland.
+     * @param bulkDataRequest includes necessary info for the bulk request
+     * @return response after posting a bulk data request to Dataland
+     *
+     */
+    @Operation(
+        summary = "Send a bulk request",
+        description = "A bulk of data requests for specific frameworks and companies is being sent.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully processed a bulk of data requests."),
+        ],
+    )
+    @PostMapping(
+        produces = ["application/json"],
+        consumes = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    fun postBulkDataRequest(
+        @Valid @RequestBody
+        bulkDataRequest: BulkDataRequest,
+    ):
+        ResponseEntity<BulkDataRequestResponse>
+
+    /** A method to get all data requests of a specific user.
+     * @return all data requests of the user in a list
+     */
+    @Operation(
+        summary = "Get data requests of a specific user.",
+        description = "Gets all the stored data request created by the user in the past.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved data requests for the user."),
+        ],
+    )
+    @GetMapping(
+        value = ["/user"],
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    fun getDataRequestsForUser(): ResponseEntity<List<DataRequestEntity>>
+
+    /** Retrieves aggregated data requests by aggregating all userIds
+     * @return aggregated data requests that match the given filters
+     */
+
+    @Operation(
+        summary = "Get aggregated data requests.",
+        description = "Gets all data requests that match the given filters, while aggregating userIDs.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved aggregated data requests."),
+        ],
+    )
+    @GetMapping(
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    fun getAggregatedDataRequests(
+        @RequestParam identifierValue: String? = null,
```

Comment:
Which identifier is expected here? Such identifiers like LEI, ISIN, PERMID?

Reply:
After a call it is clearer. DatalandId if the company is already known by dataland, otherwise one of the external identifiers like LEI, ISIN, PERMID.